src/cmap/connection.ts

```diff
@@ -264,7 +320,7 @@ function messageHandler(conn: any) {
       }
 
       if (document.$clusterTime) {
-        conn[kClusterTime] = document.$clusterTime;
+        conn[kClusterTime] = document.$clusterTime as Document;
```

Comment:
I wonder if you'd have to do this if you made it `const document: Document = message.documents[0]`. Also, maybe `message.documents` needs to be typed as `Document[]`